.github/workflows/ci.yaml

```diff
@@ -41,60 +41,134 @@ jobs:
       - name: Create timestamp
         id: create-timestamp
         run: echo "timestamp=$(date +%Y%m%d%H%M%S)" >> $GITHUB_OUTPUT
+  
+  create-cache-key:
+    name: Create cache key
+    runs-on: ubuntu-22.04
+    container:
+      image: zingodevops/android_builder:stable
+    env:
+      RUSTUP_HOME: /root/.rustup
+    outputs:
+      cache-key: ${{ steps.create-cache-key.outputs.cache-key }}
+    steps:
+      - name: Checkout zingo-mobile
+        uses: actions/checkout@v3
+        with:
+          repository: zingolabs/zingo-mobile
+      
+      - name: Store zingolib github ref
+        run: echo "zingolib_ref=$(echo ${GITHUB_REF} | sed 's/\//\\\//g')" >> $GITHUB_ENV
+      
+      - name: Update zingo-mobile dependencies
+        run: sed -i "/^zingolib\|^zingoconfig/ s/branch = \"dev\"/rev = \"${{ env.zingolib_ref }}\"/" "rust/lib/Cargo.toml"
 
-  build_android:
-    name: Build android (${{ matrix.arch }})
+      - name: Check update was successful
+        run: if cat rust/lib/Cargo.toml | grep "^zingolib\|^zingoconfig" | grep -q "branch"; then exit 1; fi
+      
+      - name: Store Cargo version
+        run: echo $(cargo --version) >> rust/cargo_version.txt
+      
+      - name: Create cache key
+        id: create-cache-key
+        run: echo "cache-key=${{ hashFiles('rust/**/Cargo.toml', 'rust/**/Cargo.lock', 'rust/**/*.rs', 'rust/cargo_version.txt') }}" >> $GITHUB_OUTPUT
+  
+  check-build-cache:
+    name: Check build cache
+    needs: create-cache-key
+    runs-on: macos-12
+    env:
+      CACHE-KEY: ${{ needs.create-cache-key.outputs.cache-key }}
+    outputs:
+      status-x86: ${{ steps.set-output-status.outputs.status-x86 }}
+      status-x86-64: ${{ steps.set-output-status.outputs.status-x86_64 }}
+      status-armeabi-v7a: ${{ steps.set-output-status.outputs.status-armeabi-v7a }}
+      status-arm64-v8a: ${{ steps.set-output-status.outputs.status-arm64-v8a }}
+    strategy:
+      fail-fast: false
+      matrix:
+        abi: [ x86, x86_64, armeabi-v7a, arm64-v8a ]
+    continue-on-error: true
+    steps:
+      - name: Checkout zingo-mobile
+        uses: actions/checkout@v3
+        with:
+          repository: zingolabs/zingo-mobile
+      
+      - name: Check if native rust cache exists
+        id: check-build-cache
+        uses: actions/cache@v3
+        with:
+          path: android/app/src/main/jniLibs/${{ matrix.abi }}
+          key: native-rust-${{ matrix.abi }}-${{ env.CACHE-KEY }}
+          lookup-only: true
+          fail-on-cache-miss: true
```

Comment:
This seems to be causing the integration test jobs to also fail due to the cache miss...is this intended?

Reply:
Are you sure this is failing integration tests? The cache is designed to fail when the rust has changed which triggers the build job

Reply:
Looking at the logs it is the build that is failing. cI is working as intended but there is errors in the rust build